packages/fuels-programs/src/contract/loader.rs

```diff
@@ -0,0 +1,314 @@
+            op::bsiz(0x11, 0x10),
+            // Push the blob contents onto the stack.
+            op::ldc(0x10, 0, 0x11, 1),
+            // Move on to the next blob.
+            op::addi(0x10, 0x10, BLOB_ID_SIZE),
+            // Decrement the loop counter.
+            op::subi(0x13, 0x13, 1),
+            // Jump backwards (3+1) instructions if the counter has not reached 0.
+            op::jnzb(0x13, RegId::ZERO, 3),
+            // 2. Jump into the memory where the contract is loaded.
+            // What follows is called _jmp_mem by the sway compiler.
+            // Subtract the address contained in IS because jmp will add it back.
+            op::sub(0x16, 0x16, RegId::IS),
+            // jmp will multiply by 4, so we need to divide to cancel that out.
+            op::divi(0x16, 0x16, 4),
+            // Jump to the start of the contract we loaded.
+            op::jmp(0x16),
+        ]
+    };
+
+    let num_of_instructions = u16::try_from(get_instructions(0, 0).len())
+        .expect("to never have more than u16::MAX instructions");
+
+    let num_of_blobs = u32::try_from(blob_ids.len()).map_err(|_| {
+        error!(
+            Other,
+            "the number of blobs ({}) exceeds the maximum number of blobs supported: {}",
+            blob_ids.len(),
+            u32::MAX
+        )
+    })?;
+
+    let instruction_bytes = get_instructions(num_of_instructions, num_of_blobs)
+        .into_iter()
+        .flat_map(|instruction| instruction.to_bytes());
+
+    let blob_bytes = blob_ids.iter().flatten().copied();
+
+    Ok(instruction_bytes.chain(blob_bytes).collect())
+}
+
+#[derive(Debug, Clone)]
+pub struct BlobsUploaded {
+    blob_ids: Vec<BlobId>,
+}
+
+#[derive(Debug, Clone)]
+pub struct BlobsNotUploaded {
+    blobs: Vec<Blob>,
+}
+
+#[derive(Debug, Clone)]
+pub struct Loader<Blobs> {
+    as_blobs: Blobs,
+}
+
+impl Contract<Loader<BlobsNotUploaded>> {
+    pub fn code(&self) -> Vec<u8> {
+        let ids: Vec<_> = self.blob_ids();
+        loader_contract_asm(&ids)
+            .expect("a contract to be creatable due to the check done in loader_for_blobs")
+    }
+
+    pub fn contract_id(&self) -> ContractId {
+        self.compute_roots().0
+    }
+
+    pub fn code_root(&self) -> Bytes32 {
+        self.compute_roots().1
+    }
+
+    pub fn state_root(&self) -> Bytes32 {
+        self.compute_roots().2
+    }
+
+    fn compute_roots(&self) -> (ContractId, Bytes32, Bytes32) {
+        compute_contract_id_and_state_root(&self.code(), &self.salt, &self.storage_slots)
+    }
+
+    /// Creates a loader contract for the code found in `blobs`. Calling `deploy` on this contract
+    /// does two things:
+    /// 1. Uploads the code blobs.
+    /// 2. Deploys the loader contract.
+    ///
+    /// The loader contract, when executed, will load all the given blobs into memory and delegate the call to the original contract code contained in the blobs.
+    pub fn loader_for_blobs(
```

Comment:
For me it is more natural to call the function `loader_from_blobs`. 

```suggestion
    pub fn loader_from_blobs(
```

Reply:
I'm also leaning towards 'for' (that's why that was the first version) 

But the leader does incorporate blobs (at least blob ids) in its code, so it is also made from the blob (ids). 

I decided to accept @hal3e suggestion because somebody might discover it more easily since there are a lot of `from_xyz` in the wild.